crates/runtime/src/lib.rs

```diff
@@ -224,13 +232,38 @@ impl Runtime {
         }
     }
 
+    fn datasets_iter(app: &App) -> impl Iterator<Item = Result<Dataset>> + '_ {
+        app.datasets.iter().cloned().map(Dataset::try_from)
+    }
+
+    /// Returns a list of valid datasets from the given App, skipping any that fail to parse and logging an error for them.
+    fn get_valid_datasets(app: &App, log_failures: bool) -> Vec<Dataset> {
```

Comment:
This is a new function that tries to parse the datasets from the spicepod into the new representation. If it fails for any reason, it skips it and returns only the valid ones. It also logs an error depending on who called it.